File: AssetInsight/src/main/java/br/unipar/assetinsight/dtos/requests/OrdemServicoRequest.java
package br.unipar.assetinsight.dtos.requests;

import java.io.Serializable;

import br.unipar.assetinsight.entities.OrdemServicoEntity;
import br.unipar.assetinsight.enums.StatusOrdemServicoEnum;
import jakarta.validation.constraints.*;

import java.sql.Timestamp;
import java.util.List;

/**
 * DTO para requests post de {@link OrdemServicoEntity}
 */
public record OrdemServicoRequest(
        Long id,
        String descricao,
        Timestamp data,
        List<ServicoRequest> servicos,
        @Positive double valorTotal
) implements Serializable {}

File: Frontend/src/app/dtos/requests/OrdemServico.request.ts
import {ServicoRequest} from "./Servico.request";

export class OrdemServicoRequest {
  id?: number;
  descricao?: string;
  data?: Date | string;
  servicos?: ServicoRequest[];

  constructor() {}

  setValues(id?: number, descricao?: string, data?: Date, servicos?: ServicoRequest[]) {
    this.id = id;
    this.descricao = descricao;
    this.data = data;
    this.servicos = servicos;
  }
}